couchdb-audit/log.js

```diff
@@ -139,18 +139,44 @@ module.exports = {
     }
 
     function get(docId, callback) {
-      getAll([docId], function(err, result) {
+      getAll([getAuditId(docId)], function(err, result) {
         callback(err, result && result.length && result[0]);
       });
     }
 
-    function getAll(docIds, callback) {
-      var ids = docIds.map(getAuditId);
-      auditDb.allDocs({ include_docs: true, keys: ids }, function(err, result) {
+    var MULTI_DOC_BATCH = 101;
+    function getAll(docIds, callback, results) {
+      results = results || {rows: []};
+
+      if (!docIds || !docIds.length) {
+        return callback(null, results);
+      }
+
+      var batch = docIds.splice(docIds.length - MULTI_DOC_BATCH);
+      auditDb.allDocs({ include_docs: true, keys: batch }, function(err, batchResult) {
+        if (err) {
+          return callback(err);
+        }
+
+        results.rows = results.rows.concat(batchResult.rows);
+
+        getAll(docIds, callback, results);
```

Comment:
I'm not comfortable with the recursion here. It's hard to read and not memory efficient. Seeing async is already a dependency, how about using async.map or something? I feel like there should be an async.batch...